### PR TITLE
[CORE-567] add integration tests for sending module governance messages

### PR DESCRIPTION
**Pull request by tqin7**
add integration tests for governance message `MsgSendFromModuleToAccount`

### Summary by CodeRabbit

- New Feature: Added a new test function `TestSendFromModuleToAccount` in `protocol/testing/e2e/gov/sending_test.go`. This function tests the functionality of sending coins from a module to an account, covering scenarios such as successful transfers, insufficient balance, and invalid authority. This will help ensure the robustness of our coin transfer feature and catch potential issues early.

### CODE REVIEW

**Review by ttl33 — APPROVED** [review]

File: protocol/testing/e2e/gov/sending_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				SenderModuleName: vesttypes.CommunityTreasuryAccountName,
+				Recipient:        constants.AliceAccAddress.String(),
+				Coin:             sdk.NewCoin("dv4tnt", sdk.NewInt(123)),
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Fail: invalid authority": {
+			msg: &sendingtypes.MsgSendFromModuleToAccount{
+				Authority:        authtypes.NewModuleAddress(sendingtypes.ModuleName).String(),
+				SenderModuleName: banktypes.ModuleName,
```
[the remark this answers is not in the document]
Comment (tqin7) [reply protocol/testing/e2e/gov/sending_test.go:37]: Agreed!

File: protocol/testing/e2e/gov/sending_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	tests := map[string]struct {
+		msg                      *sendingtypes.MsgSendFromModuleToAccount
+		expectedProposalStatus   govtypesv1.ProposalStatus
+		expectSubmitProposalFail bool
+	}{
+		"Success": {
+			msg: &sendingtypes.MsgSendFromModuleToAccount{
+				Authority:        authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				SenderModuleName: vesttypes.CommunityTreasuryAccountName,
+				Recipient:        constants.AliceAccAddress.String(),
```
[the remark this answers is not in the document]
Comment (tqin7) [reply protocol/testing/e2e/gov/sending_test.go:29]: I was thinking about keeping these integration tests lightweight as unit tests already comprehensively cover different scenarios. Mainly thinking of having these integration tests as a sanity check on governance flow. Let me know what you think!

[the remark this answers is not in the document]
Comment (ttl33) [reply protocol/testing/e2e/gov/sending_test.go:29]: posting here for the record:

I see. Is there a downside to adding these test cases other than the time it takes to run the tests?
If these tests are not flaky, I think it's worth adding them for completeness sake

File: protocol/testing/e2e/gov/sending_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+				Coin:             sdk.NewCoin("dv4tnt", sdk.NewInt(123)),
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Fail: invalid authority": {
```
[the remark this answers is not in the document]
Comment (tqin7) [reply protocol/testing/e2e/gov/sending_test.go:34]: 1. will add!
2. currently if sender is not a module account, `bankKeeper` will panic. this is not the ideal behavior as gov module endblocker currently doesn't recover from panic (fixed in 0.50.0 i believe)
3. if recipient doesn't exist, cosmos will simply create that recipient address

**Pull request merged**